externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala

```diff
@@ -71,41 +71,48 @@ case class SparkSQLEngine(spark: SparkSession) extends Serverable("SparkSQLEngin
 
 object SparkSQLEngine extends Logging {
 
-  val sparkConf: SparkConf = new SparkConf()
+  private var _sparkConf: SparkConf = _
 
-  val kyuubiConf: KyuubiConf = KyuubiConf()
+  private var _kyuubiConf: KyuubiConf = _
+
+  def kyuubiConf: KyuubiConf = _kyuubiConf
 
   var currentEngine: Option[SparkSQLEngine] = None
 
-  private val user = currentUser
+  private lazy val user = currentUser
 
   private val countDownLatch = new CountDownLatch(1)
 
+  SignalRegister.registerLogger(logger)
+  setupConf()
```

Comment:
Should the `SparkSQLEngine.setupConf()` call in `WithSparkSQLEngine#startSparkEngine` be removed?

Reply:
No, after changing system properties, we need re-init the SparkConf and KyuubiConf to let them pick up the system properties